lib/ci-stack.ts

```diff
@@ -80,7 +85,13 @@ export class CIStack extends Stack {
       oidcCredArn: importedOidcConfigValuesSecretBucketValue.toString(),
       useSsl,
       runWithOidc,
-    });
+    },
+    {
+      agentNodeSecurityGroup: securityGroups.agentNodeSG.securityGroupId.toString(),
+      subnetId: vpc.publicSubnets[0].subnetId.toString(),
+    },
+    agentNodesConfig.AL2_X64,
```

Comment:
Can you have the main node accept an agentNodesConfig and then it extracts what it needs?

Reply:
Done